File: AlgorithmTrain/Sort/InsertionSort/InsertionSort.py
def insertionSort(nums):
    n = len(nums)
    for i in range(1,n):
        j = i-1
        while j>=0 and nums[j+1]<nums[j]:
            #交換nums[j]和nums[j+1]
            nums[j],nums[j+1]=nums[j+1],nums[j] 
            j -= 1


if __name__ == "__main__":
    nums = [1, 3, 5, 2, 13, 7, 5]
    insertionSort(nums)
    print(nums)

File: AlgorithmTrain/Sort/InsertionSort/InsertionSort.java
package AlgorithmTrain.Sort.InsertionSort;

import java.util.Arrays;

public class InsertionSort {
    private static void insertionSort(int[] nums) {
        int i, j, n = nums.length;
        for (i = 1; i < n; i++) {
            int tmp = nums[i];
            for (j = i - 1; j >= 0 && tmp < nums[j]; j--) {
                nums[j + 1] = nums[j];
            }
            nums[j + 1] = tmp;
        }
    }

    public static void main(String[] args) {
        int[] nums = { 1, 3, 5, 2, 13, 7, 5 };
        insertionSort(nums);
        System.out.println(Arrays.toString(nums));
    }
}
